src/_pytest/logging.py

```diff
@@ -71,7 +71,7 @@ def formatTime(self, record: LogRecord, datefmt: Optional[str] = None) -> str:
             tz = timezone(timedelta(seconds=ct.tm_gmtoff), ct.tm_zone)
             # Construct `datetime.datetime` object from `struct_time`
             # and msecs information from `record`
-            dt = datetime(*ct[0:6], microsecond=round(record.msecs * 1000), tzinfo=tz)
+            dt = datetime(*ct[0:6], microsecond=int(record.msecs * 1000), tzinfo=tz)
```

Comment:
I'm OK with leaving this small change only for Python 3.7, but we should document it as such to avoid an accidental regression:

```suggestion
            # Using int() instead of round() to avoid it exceeding 1_000_000 and causing a ValueError (#11861).          
            dt = datetime(*ct[0:6], microsecond=int(record.msecs * 1000), tzinfo=tz)
```